pkg/authenticator/kubeconfig.go

```diff
@@ -43,6 +43,11 @@ func (kc *kubeconfigClient) GetKubeconfig(ctx context.Context, clusterName strin
 		return "", fmt.Errorf("getting kubeconfig for cluster %q: %w", clusterName, err)
 	}
 
+	if _, err := os.Stat(secretName); err == nil {
+		_ = os.Chown(secretName, os.Geteuid(), os.Getegid())
```

Comment:
I would have returned these as errors. Better to know sooner than later that there's a problem.

I'm guessing you disagree, but would you be amenable to at least a log message, so we'd have some clue that there's something wrong?

Reply:
99% sure the error would be permission denied which will show up below on the write. I don't want to clutter this up with code I think is a bit throw away away. Hopefully, we move away from this file approach.